Makefile

```diff
@@ -140,7 +140,7 @@ plugout_ldflags += -laudio $(libaudio_flags)
 endif
 ifeq ($(plugout_sdl),yes)
 plugout_objs += plugout_sdl.o
-plugout_ldflags += $(libSDL2_flags)
+plugout_ldflags += -lSDL2 $(libSDL2_flags)
```

Comment:
On my Debian the `-lSDL2` is the sole output of `sdl2-config --libs`.

In the configure script `$sdl2_lib_path` only seems to contain the path to the libs (the `-L`  part).
`$sdl2_lib_path` seems to become `$(libSDL2_flags)` here but then the rest of the variables from `sdl2-config --libs` seems to be missing.
Could we instead use the whole output for of `sdl2-config --libs`, pass it through to here and remove the fixed `-lSDL2`?

Reply:
There's also a duplicate `-lm` :)

It might be an alternative to just pass the output of "sdl2-config --libs" in its entirety. Currently I'm only calling it to get the effective library path.
This is what I get on MingW64:
```
$ sdl2-config --libs
-LD:/msys64/mingw64/lib -lmingw32 -mwindows -lSDL2main -lSDL2
```
Then `check_libs` tries a compile with the given `-L` option and adds `-lSDL2` to the `libSDL2_flags` if it's successful. On Windows the compile appears to fail either way so it never adds -lSDL2.
It's complaining about an undefined reference to `WinMain` (why...)
However it does work when all the switches from sdl2-config --libs are added (and in that precise order).

Reply:
I think `check_libs` might be the wrong tool here. I can pass the flags from `sdl2-config --libs` into `check_libs` but then it will succeed on first compile, deduce that it doesn't need to add any flags, and simply return so the flags are lost and not written to `libSDL2_flags`.

Reply:
The test snippet can be made to compile on Windows by adding `#define SDL_MAIN_HANDLED` without having to add the whole sermon of flags from sdl2_config - which posed their own problems as they kept gbsplay from launching properly, or at least from handling the terminal properly (no input, no output).